optimum/graphcore/pipelines/__init__.py

```diff
@@ -0,0 +1,341 @@
+            "ipu_config": "Graphcore/vit-base-ipu",
+        },
+        "type": "image",
+    },
+    "question-answering": {
+        "impl": QuestionAnsweringPipeline,
+        "class": (AutoModelForQuestionAnswering,),
+        "default": {
+            "model": ("distilbert-base-cased-distilled-squad", "626af31"),
+            "ipu_config": "Graphcore/distilbert-base-ipu",
+        },
+        "type": "text",
+    },
+    "text-classification": {
+        "impl": TextClassificationPipeline,
+        "class": (AutoModelForSequenceClassification,),
+        "default": {
+            "model": ("distilbert-base-uncased-finetuned-sst-2-english", "af0f99b"),
+            "ipu_config": "Graphcore/distilbert-base-ipu",
+            "padding_length": 128,
+        },
+        "type": "text",
+    },
+    "token-classification": {
+        "impl": IPUTokenClassificationPipeline,
+        "class": (AutoModelForTokenClassification,),
+        "default": {
+            "model": ("dbmdz/bert-large-cased-finetuned-conll03-english", "f2482bf"),
+            "ipu_config": "Graphcore/bert-large-ipu",
+            "padding_length": 128,
+        },
+        "type": "text",
+    },
+    "zero-shot-classification": {
+        "impl": IPUZeroShotClassificationPipeline,
+        "class": (AutoModelForSequenceClassification,),
+        "default": {
+            "model": ("roberta-large-mnli", "130fb28"),
+            "ipu_config": "Graphcore/roberta-large-ipu",
+            "padding_length": 128,
+        },
+        "type": "text",
+    },
+}
+
+NO_FEATURE_EXTRACTOR_TASKS = set()
+NO_TOKENIZER_TASKS = set()
+for task, values in SUPPORTED_TASKS.items():
+    if values["type"] == "text":
+        NO_FEATURE_EXTRACTOR_TASKS.add(task)
+    elif values["type"] in {"audio", "image"}:
+        NO_TOKENIZER_TASKS.add(task)
+    elif values["type"] != "multimodal":
+        raise ValueError(f"SUPPORTED_TASK {task} contains invalid type {values['type']}")
+
+
+def get_poplar_executor(model: PreTrainedModel, ipu_config: Union[str, dict] = None, fp16: bool = True):
+    if isinstance(ipu_config, str):
+        ipu_config = IPUConfig.from_pretrained(ipu_config)
+    elif isinstance(ipu_config, dict):
+        ipu_config = IPUConfig.from_dict(ipu_config)
+    else:
+        raise ValueError("ipu_config must be a string or a dictionary.")
+    ipu_config.inference_device_iterations = 1
+    # TODO: inference_replication_factor should be adaptive, especially for batching.
+    ipu_config.inference_replication_factor = 1
+    model = to_pipelined(model, ipu_config, force=False)
+    model.parallelize()
+    if fp16:
+        model.half()
+    opts = ipu_config.to_options(for_inference=True)
+    opts.setExecutionStrategy(poptorch.ShardedExecution())
+    model = poptorch.inferenceModel(model.eval(), opts)
+    return model
+
+
+def get_inference_context(self):
+    return torch.no_grad
+
+
+def check_model_type(self, supported_models: Union[List[str], dict]):
```

Comment:
Consistent type annotation, I guess to match the rest of the library:
```suggestion
def check_model_type(self, supported_models: Union[List[str], Dict[Type, Type]]):
```
(just guessing here for the "`Type`" part.

Reply:
I think this is a reasonable change. However, my `check_model_type` is just a function override, so I think it is wise to keep the change minimal and leave the other changes to the `transformers` library.